src/components/SchedulePanel/components/OfficeHoursDialog/index.js

```diff
@@ -0,0 +1,526 @@
+          checkedTh: nextProps.resourceId === 5 ? true : false,
+          checkedFr: nextProps.resourceId === 6 ? true : false,
+          checkedSa: nextProps.resourceId === 7 ? true : false,
+        },
+        startHourInput:nextProps.startTime,
+        endHourInput:nextProps.endTime,
+        checkedValid: true,
+        startHourValid: true,
+        endHourValid: true,
+      });
+    }
+  }
+
+  validateField(fieldName, value) {
+    let fieldValidationErrors = this.state.formErrors;
+    let descriptionValid = this.state.descriptionValid;
+    let locationValid = this.state.locationValid;
+    let startHourValid = this.state.startHourValid;
+    let endHourValid = this.state.endHourValid;
+
+    // Require that content havint appropriate size and format
+    switch (fieldName) {
+      case 'descriptionInput':
+        descriptionValid = value.trim() !== '' && value.length < 50;
+        fieldValidationErrors.descriptionInput = descriptionValid ? '' : 'Description is not Valid';
+        break;
+      case 'locationInput':
+        locationValid = value.trim() !== '' && value.length < 50;
+        fieldValidationErrors.locationInput = locationValid ? '' : 'Location is not Valid';
+        break;
+      case 'startHourInput':
+        startHourValid =
+          value.trim() !== '' &&
+          parseInt(this.state.endHourInput, 10) > parseInt(value, 10) &&
+          parseInt(value, 10) >= 5 &&
+          parseInt(value, 10) <= 10;
+        fieldValidationErrors.startHourInput = startHourValid ? '' : 'Start Time is not Valid';
+        break;
+      case 'endHourInput':
+        endHourValid =
+          value.trim() !== '' &&
+          parseInt(this.state.startHourInput, 10) < parseInt(value, 10) &&
+          parseInt(value, 10) >= 5 &&
+          parseInt(value, 10) <= 22;
+        fieldValidationErrors.endHourInput = endHourValid ? '' : 'End Time is not Valid';
+        break;
+      default:
+        break;
+    }
+    this.setState(
+      {
+        formErrors: fieldValidationErrors,
+        descriptionValid: descriptionValid,
+        locationValid: locationValid,
+        startHourValid: startHourValid,
+        endHourValid: endHourValid,
+      },
+      () => {
+        this.validateForm();
+      },
+    );
+  }
+
+  validateForm() {
+    this.setState({
+      formValid:
+        this.state.locationValid &&
+        this.state.descriptionValid &&
+        this.state.startHourValid &&
+        this.state.endHourValid &&
+        this.state.checkedValid,
+    });
+    console.log("Form Valid ", this.state.formValid);
+  }
+
+  handleSubmit = e => {
+    e.preventDefault();
+
+    var startHour = this.state.startHourInput;
+    var endHour = this.state.endHourInput;
+    var location = this.state.locationInput;
+    var description = this.state.descriptionInput;
+
+    var mySchedule = {
+      startHour: startHour,
+      endHour: endHour,
+      location: location,
+      description: description,
```

Comment:
We do you declare these as variables based on the state, but take the next values directly from state? Can we just retrieve all these values directly from the state?

Reply:
Change made